ocaml/parsing/pprintast.ml

```diff
@@ -569,47 +564,78 @@ and pattern_jane_syntax ctxt attrs f (pat : Jane_syntax.Pattern.t) =
       (attributes ctxt) attrs
   else
     match pat with
+    | Jpat_local (Lpat_local p) ->
+        pp f "(local_ %a)" (simple_pattern ctxt) p
     | Jpat_immutable_array (Iapat_immutable_array l) ->
         pp f "@[<2>[:%a:]@]"  (list (pattern1 ctxt) ~sep:";") l
     | Jpat_unboxed_constant c -> unboxed_constant ctxt f c
 
-and maybe_local_pat ctxt is_local f p =
-  if is_local then
-    pp f "(local_ %a)" (simple_pattern ctxt) p
-  else
-    pp f "%a" (simple_pattern ctxt) p
-
 and label_exp ctxt f (l,opt,p) =
```

Comment:
I didn't review the changes to `label_exp` too closely, but I'm surprised how many of them there are. Feel free to resolve this comment if you're confident that existing tests (plus your testing of the Jane Street codebase against this PR) makes you believe that these nontrivial changes to `Pprintast` are correct. (But I do wonder whether all of these changes are needed...)